simpledotorg/simple-server
# Empty seed type and better blocks

**Story card:** [ch4205](https://app.clubhouse.io/simpledotorg/story/4205/optionally-disable-patient-data-generation-in-the-seed-generation-script)

## Because

The mobile team would like to spin up and spin down server instance rapidly to back their automated test suites. Re-using the QA environment over and over again is not great ([see epic](https://app.clubhouse.io/simpledotorg/epic/4222/isolate-android-integration-tests)).

## This addresses

Two big enhancements to our seeds to support mobile tests:
* Adds an `empty` seed type that creates facilities, but doesn't populate any patient data. The number of facilities, etc, matches the `small` seed type.
* Creates more than one block per facility group. Taking a look at the old code, it seems like we intended to create several blocks per facility group, but left those mechanisms unused. We ended up creating only 1 block per FG, and the mobile tests need at least one FG with two blocks.

## Test instructions

Empty seed type
* Reset your local DB
* Run `SEED_TYPE=empty rails db:seed`
* Facilities should be created, patient data should not

Environment-based check
* Reset your local DB
* Set `SIMPLE_SERVER_ENV=android_review`
* Run `rails db:seed`
* Facilities should be created, patient data should not

Block check
* Reset your local DB
* Run `rails db:seed`
* Facilities should be created, patient data should be created, most if not all facility groups should have more than one block
